common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -208,37 +208,29 @@ public static void load(String name, ClassLoader loader) {
                 out.write(buffer, 0, length);
             }
             out.flush();
-            out.close();
-            out = null;
 
             System.load(tmpFile.getPath());
-            loaded = true;
         } catch (Exception e) {
             throw (UnsatisfiedLinkError) new UnsatisfiedLinkError(
                     "could not load a native library: " + name).initCause(e);
         } finally {
-            if (in != null) {
-                try {
-                    in.close();
-                } catch (IOException ignore) {
-                    // ignore
-                }
-            }
-            if (out != null) {
-                try {
-                    out.close();
-                } catch (IOException ignore) {
-                    // ignore
-                }
+            safeClose(in);
+            safeClose(out);
+            // Once we loaded the library its safe to also delete the file.
```

Comment:
`Once we loaded the library its safe to also delete the file.` -> `After we load the library it is safe to delete the file.`